applications/crossbar/src/modules_v2/cb_users_v2.erl

```diff
@@ -518,6 +520,40 @@ check_user_schema(UserId, Context) ->
     OnSuccess = fun(C) -> on_successful_validation(UserId, C) end,
     cb_context:validate_request_data(<<"users">>, Context, OnSuccess).
 
+-spec check_hotdesk_id(kz_term:api_binary(), cb_context:context()) -> cb_context:context().
+check_hotdesk_id(UserId, Context) ->
+    JObj = cb_context:req_data(Context),
+    HotdeskId = kz_json:get_ne_value([<<"hotdesk">>, <<"id">>], JObj),
+    AccountDb = cb_context:account_db(Context),
+    case is_hotdesk_id_unique(AccountDb, UserId, HotdeskId) of
+        'true' ->
+            Context;
+        'false' ->
+            lager:debug("Hotdesk.id ~p is already used", [HotdeskId]),
+            non_unique_hotdesk_id_error(Context, HotdeskId);
+        {'error', DatamgrError} ->
+            crossbar_doc:handle_datamgr_errors(DatamgrError, HotdeskId, Context)
+    end.
+
+-spec is_hotdesk_id_unique(kz_term:api_binary(), kz_term:api_binary(), kz_term:ne_binary()) -> boolean() | {'error', any()}.
+is_hotdesk_id_unique(AccountDb, UserId, HotdeskId) ->
+    ViewOptions = [{'key', HotdeskId}],
+    case kz_datamgr:get_results(AccountDb, ?LIST_BY_HOTDESK_ID, ViewOptions) of
+        {'ok', []} -> 'true';
+        {'ok', [JObj|_]} -> kz_doc:id(JObj) =:= UserId;
+        Else ->
+            lager:debug("error ~p checking view ~p in ~p", [Else, ?LIST_BY_HOTDESK_ID, AccountDb]),
+            {'error', Else}
```

Comment:
Won't `Else` be `{error, Something}`? So won't this become `{error, {error, Something}}`?

Reply:
Good catch!